contribute/design/1.x/autoscale/cstor/20191014-cstor-autoscale.md

```diff
@@ -0,0 +1,339 @@
+`"cluster-autoscaler.kubernetes.io/safe-to-evict": "false"` as annotation on
+cStor pool related pods.
+
+#### Adding new node to existing CSPC
+
+Either admin (or) operator which recognizes the need to add new pool to
+existing CSPC adds a `node entry` to above yaml.
+cspc-operator then takes care of creating required resources to add new pool to
+existing pool cluster.
+
+#### Informing that a pool has been `cordoned` to the system
+
+This is required to stop provisioning new replicas on a pool that eventually
+gets deleted from pool cluster.
+
+A new field will be added to CSPI which can be read by CVC controller during
+the phase of volume provision.
+This being related to provision, field for this in CSPI is:
+`provision.status`. This can take `ONLINE` and `CORDONED` as possible values.
+
+`ONLINE` means CVC can pick the CSPI for volume provisioning.
+
+`CORDONED` means CVC SHOULD NOT pick the CSPI for any further volume
+provisioning.
+
+There is NO reconcilation happens for this, but, this acts as a config
+parameter.
+
+This is not added to `spec` of CSPI as `spec` is shared with CSPC controller
+also.
+
+#### Draining the `cordoned` pool
+
+Admin (or) operator drains a pool (DP) by moving replicas which are on DP to
+another pool in the same cluster. This can be in two modes.
+
+One is - by adding new replica and deleting old one
+
+Second is - by moving replica from old one to new replica
+
+Admin (or) operator need to identify the list of replicas i.e., CVRs on pool DP.
+For each identified CVR (CVR1), follow either the first mode or second.
+
+__Add a new replica and remove old one__ way:
+- Identify new pool (NP) other than DP which can take the replica
+- Perform replica scaleup steps
+  - create a new CVR (CVR2) on NP
+  - Increase the desired replication factor by 1 in CV CR related to this volume.
+- Once the CVR2 becomes healthy, perform replica scaledown steps
+  - remove CVR1 related details, and, reduce DRF on CV CR
+  - delete CVR1
+
+Above steps to add new replica are given in
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/replica_scaleup/dataplane/20190910-replica-scaleup.md
+
+Steps to perform replica scale down can change as its implementation starts.
+
+__Move old replica to new one__ way:
+- Identify new pool (NP) other than DP which can take the replica
+- Delete CVR1 which is on old pool
+- Create new CVR (CVR2) on new pool with status as 'Recreate' and
+`spec.ReplicaID` same as that of CVR1
+
+Above steps to move replica are also available at
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/replica_scaleup/dataplane/20190910-replica-scaleup.md
+
+#### Movement of pools on remote disks from one node to another
+
+This scenario is about moving pools on remote disks from old node to another,
+instead of distributing replicas on pool of old node to other pools.
+
+Steps to perform are as follows:
+- Identify the node N1 on which replicas of old pool doesn't exists
```

Comment:
Can we ignore this step? Are we supporting distributing volume replicas across different CSPC?
It leads to another way of understanding consider below case
1. Four node cluster and created CSPI(cStor pools) on all four nodes.
2. I have only single replica volumes in all the pools. (This point leads to the assumption of moving cspi1 to node3 which has cspi3).